src/main/java/fi/aalto/cs/apluscourses/presentation/AuthenticationViewModel.java

```diff
@@ -48,14 +58,24 @@ public Authentication build() {
       if (token == null) {
         throw new IllegalStateException("Token is not set");
       }
-      Authentication authentication = authenticationFactory.create(token);
+      this.authentication = authenticationFactory.create(token);
       Arrays.fill(token, '\0');
-      return authentication;
+      return this.authentication;
```

Comment:
I suggest that this method is made void typed to emphasize that the object that is built is the one stored into this viewmodel.